python/lsst/ts/LinearStage/csc.py

```diff
@@ -1,91 +1,275 @@
+
+    def __init__(
+        self,
+        index,
+        initial_state=salobj.State.STANDBY,
+        config_dir=None,
+        simulation_mode=0,
+    ):
+        schema_path = (
+            pathlib.Path(__file__)
+            .resolve()
+            .parents[4]
+            .joinpath("schema", "LinearStage.yaml")
+        )
+        super().__init__(
+            name="LinearStage",
+            index=index,
+            schema_path=schema_path,
+            config_dir=config_dir,
+            initial_state=initial_state,
+            simulation_mode=simulation_mode,
+        )
+
+        self.evt_detailedState.set_put(
+            detailedState=LinearStage.DetailedState(
+                LinearStage.DetailedState.NOTMOVINGSTATE
+            )
+        )
+
+        self.component = None
+        self.referenced = False
+        self.telemetry_task = salobj.make_done_future()
+        self.simulation_mode_number = simulation_mode
+        self.log.debug(
+            f"LinearStage CSC initialized, simulation number is set to {self.simulation_mode_number}"
+        )
+
+    @staticmethod
+    def get_config_pkg():
+        """Return the name of the configuration package."""
+        return "ts_config_mtcalsys"
+
+    async def configure(self, config):
+        """Configure the CSC.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+        """
+        self.stage_type = config.stage_type
+
+        self.log.debug(f"Stage type is {self.stage_type}")
+        self.log.debug(f"Simulation mode number is {self.simulation_mode_number}")
+        # Instantiate the class specific to the hardware component
+        if self.stage_type == "Igus":
+            self.component = IgusLinearStageStepper(
+                simulation_mode=bool(self.simulation_mode_number), log=self.log
+            )
+        elif self.stage_type == "Zaber":
+            self.component = ZaberLSTStage(
+                simulation_mode=bool(self.simulation_mode_number), log=self.log
+            )
+
+        else:
+            raise IOError("Stage type not defined in config file.")
+
+        self.component.configure(config)
+
+    @property
+    def detailed_state(self):
+        """The substate of the LinearStage.
+
+        Parameters
+        ----------
+        new_sub_state : `LinearStage.DetailedState`
+
+        Returns
+        -------
+        LinearStage.DetailedState
+            The current sub state of the LinearStage.
+        """
+        return LinearStage.DetailedState(self.evt_detailedState.data.detailedState)
+
+    @detailed_state.setter
+    def detailed_state(self, new_sub_state):
+        new_sub_state = LinearStage.DetailedState(new_sub_state)
+        self.evt_detailedState.set_put(detailedState=new_sub_state)
+
+    def assert_referenced(self, action):
+        # Stage must be homed/referenced before attempting
+        # an absolute positioning
+        if not self.referenced:
+            raise AssertionError(

```

Comment:
Can you raise ˋsalobj.ExpectedErrorˋ instead? Simply because that's what happens in ˋassert_enabledˋ.

Reply:
I can, but is the proper fix to change what's in `assert_enabled` or to change it here?
As far as I can tell `salobj.ExpectedError` just doesn't provide a traceback whereas this error will?
I agree things should be consistent... just not sure I understand which is the best choice. Any thoughts?

Reply:
@r-owen can probably provide a better explanation and opinion here.

Reply:
@couger01 's explanation makes sense to me. As you have both said: it's purely a question of whether a traceback is any use.